stl/inc/variant

```diff
@@ -1611,14 +1646,31 @@ constexpr _Variant_visit_result_t<_Callable, _As_variant<_Variants>...> visit(_C
         _Meta_transform<_Meta_quote<_Meta_as_integer_sequence>, _Meta_cartesian_product<_ListOfIndexLists>>;
     static_assert(_Variant_all_visit_results_same<_Callable, _ListOfIndexVectors, _As_variant<_Variants>...>::value,
         "visit() requires the result of all potential invocations to have the same type and value category "
-        "(N4828 [variant.visit]/2).");
+        "(N4835 [variant.visit]/2).");
 
-    constexpr int _Strategy =
-        _Size == 1 ? 0 : _Size <= 4 ? 1 : _Size <= 16 ? 2 : _Size <= 64 ? 3 : _Size <= 256 ? 4 : -1;
-    return _Visit_strategy<_Strategy>::template _Visit<_ListOfIndexVectors>(
-        _Variant_visit_index1(0, static_cast<_As_variant<_Variants>&>(_Args)...), static_cast<_Callable&&>(_Obj),
-        static_cast<_As_variant<_Variants>&&>(_Args)...);
+    return _Visit_impl<_Size, _ListOfIndexVectors, _Variant_visit_result_t<_Callable, _As_variant<_Variants>...>>(

```

Comment:
Would this be clearer if you extracted `using _Ret = _Variant_visit_result_t<_Callable, _As_variant<_Variants>...>;`?